### ISSUE TITLE
lambda command

### ISSUE THREAD

**Issue by simonw** [simonw/sqlite-transform#2]
This will allow users to specify their own Python expression or multi-line expression for transforming columns.

For example:

    $ sqlite-transform lambda mydb.db mytable col1 --expr='value.upper()'

**Comment by simonw** [comment]
I think I'm going to have the code come as a required option rather than a positional argument because mixing in that code with the database file and table and columns could get a little messy.

**Comment by simonw** [comment]
This example is fine for single expression transformations, but it would be great if we could support multi-line function bodies as well. Maybe something like this:

    $ sqlite-transform lambda mydb.db mytable col1 --body='
    upper = value.upper()
    return upper.replace("foo", "bar")'

This would be automatically compiled as the body of a function that takes a single `value` argument.

**Comment by simonw** [comment]
For both of these variants we can also support multiple `--import=` arguments for specifying modules to import.

I mucked around a bit and eventually came up with this recipe for handling dynamic imports with code that is compiled using Python's `compile()` function.

```python
code = compile("""
def fn(url): 
    return requests.get(url).json() 
""", "<string>", "exec")
locals = {}
globals = {"requests": __import__("requests")}
exec(code, globals, locals)
fn = locals['fn']
json_data = fn("https://latest.datasette.io/.json")
```

**Comment by simonw** [comment]
I'm going to ditch `--expr` and rename `--body` above to `--code`. If you want a single line transformation you'll specify it like this:

    $ sqlite-transform lambda mydb.db mytable col1 --code='return value.upper()'

This feels less confusing than having two commands depending on if you are using more than one statement or not.

**Comment by simonw** [comment]
Released in `0.2` - documentation here: https://github.com/simonw/sqlite-transform/blob/0.2/README.md#lambda-for-executing-your-own-code
